lib/floe/workflow/state.rb

```diff
@@ -25,21 +25,42 @@ def initialize(workflow, name, payload)
         @workflow = workflow
         @name     = name
         @payload  = payload
+        # All states must define Next or End except for Choice, Succeed, and Fail
         @end      = !!payload["End"]
+        @next     = payload["Next"]
+        # All states must define a Type
         @type     = payload["Type"]
```

Comment:
If we want to ensure it, then we can change this to the following (and then then comment is redundant)

```suggestion
        # All states must define a Type
        @type     = payload.fetch("Type")
```